client/jetpack-connect/index.js

```diff
@@ -114,11 +100,10 @@ export default function () {
 			page.redirect( `/jetpack/connect/store${ params.interval ? '/' + params.interval : '' }` )
 	);
 
-	if ( isLoggedOut ) {
-		page( '/jetpack/connect/plans/:interval(yearly|monthly)?/:site', ( { path } ) =>
-			page( login( { isNative: true, isJetpack: true, redirectTo: path } ) )
-		);
-	}
+	page(
+		'/jetpack/connect/plans/:interval(yearly|monthly)?/:site',
+		controller.redirectToLoginIfLoggedOut
+	);
```

Comment:
Could this be merged into the `jetpackPlans` handler, so that the route is handled all-inclusive at one place?

The site-specific `/jetpack/connect/store/yearly/example.blog` route, also set up with `jetpackPlans`, should probably also redirect to login, but it currently doesn't.